File: exercises/src/balanced.js
const balanced = (letters) => {
  if (!letters) {
    return true;
  }

  const characters = letters.split("");
  const xCount = characters.filter((character) => character === "x").length;
  const yCount = characters.filter((character) => character === "y").length;

  return xCount === yCount;
};

export default balanced;

File: exercises/src/balanced.rs
pub fn is_balanced(letters: &str) -> bool {
    let size = letters.chars().count();

    match size {
        0 => true,
        1 => false,
        _ => {
            let x_count = letters.chars().filter(|letter| letter.eq(&'x')).count();
            let y_count = letters.chars().filter(|letter| letter.eq(&'y')).count();

            x_count == y_count
        }
    }
}
